Update cloud cli docs with windows, new instructions (#4172)
fishtown-analytics/docs.getdbt.com: website/docs/docs/cloud/configure-cloud-cli.md

1. Ensure you meet the prerequisites above.
2. Create an environment variable with your [dbt Cloud API key](/docs/dbt-cloud-apis/user-tokens):
   - On MacOS, Linux, or Windows add an environment variable:

        ```bash
        export DBT_CLOUD_API_KEY="1234" # Replace 1234 with your API key   
        ```

   - In Powershell, add an environment variable: IS THIS MISSING SOMETHING?
     - Note that this variable resets if you restart your shell. To add an environment variable permanently, add a system environment variable in your platform.

3. Navigate to a dbt project in your terminal:

```bash
cd ~/dbt-projects/jaffle_shop
```

4. In your `dbt_project.yml` file, ensure there is a section titled `dbt-cloud`. This section is required to have a `project-id` field with a valid project ID. 

```yaml
# dbt_project.yml
name:

version:
...

dbt-cloud: 
    project-id: PROJECT_ID
```

- To find your project ID, go to **Develop** in the navigation menu. Select the dbt Cloud project URL, such as `https://cloud.getdbt.com/develop/26228/projects123456`, where the project ID is `123456`.

## Use the dbt Cloud CLI

> The dbt Cloud CLI shares the same set of commands as dbt Core. When you invoke a dbt command, that command is sent to dbt Cloud for processing. Since this is still invoking dbt under the hood, you'll still need to run `dbt deps`, followed by the same model build commands you typically run.

Reviewer: ```suggestion
The dbt Cloud CLI shares the same set of commands as dbt Core. When you invoke a dbt command, that command is sent to dbt Cloud for processing. 
```

Reviewer: will add info about dbt commands

Reviewer: add a note for user to run `dbt deps` if they change a package version or remove a package

---
title: Configure dbt Cloud CLI
id: configure-cloud-cli

> description: "Instructions on how to configure the dbt Cloud CLI"

Reviewer 2: The "next" button currently links to the IDE docs because it's the next category. Do we want this behavior?

---

> :::info Public preview functionality

Reviewer 2: Same feedback: can this be a reusable snippet?

The dbt Cloud CLI is currently in [public preview](/docs/dbt-versions/product-lifecycles#dbt-cloud). Share feedback or request features you'd like to see on the [dbt community Slack](https://getdbt.slack.com/archives/C05M77P54FL).

::: 

## Prerequisites

- You must set up a project in dbt Cloud.
- You must have your [personal development credentials](/docs/dbt-cloud-environments#set-developer-credentials) set for that project. The dbt Cloud CLI will use these credentials, stored securely in dbt Cloud, to communicate with your data platform.
- You must [enroll](/docs/dbt-versions/experimental-features) in the dbt Cloud beta features. 
	- To enroll, navigate to your **Profile Settings** and enable the **Beta** flag under **Experimental Features**.

## Configure the dbt Cloud CLI

Once you install the dbt Cloud CLI, you need to configure it to connect to a dbt Cloud project.

1. Ensure you meet the prerequisites above.
2. Create an environment variable with your [dbt Cloud API key](/docs/dbt-cloud-apis/user-tokens):
   - On MacOS, Linux, or Windows add an environment variable:

        ```bash
        export DBT_CLOUD_API_KEY="1234" # Replace 1234 with your API key   
        ```

> 

Reviewer 2: Form line 34 down the text turns blue in GitHub. I think this means there's a space or code error before it. It looks fine in the output, so maybe nbd

Reviewer: thank you Leona! looks like Greg made some updates

>    - In Powershell, add an environment variable: IS THIS MISSING SOMETHING?

Reviewer 2: I'm not a Windows user, but we can point them to these docs maybe 🤷🏻 
```suggestion
  - In Powershell, add an environment variable. Refer to [About environment variables](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_environment_variables?view=powershell-7.3) for more information.
```

     - Note that this variable resets if you restart your shell. To add an environment variable permanently, add a system environment variable in your platform.

3. Navigate to a dbt project in your terminal:

```bash
cd ~/dbt-projects/jaffle_shop
```

> 4. In your `dbt_project.yml` file, ensure there is a section titled `dbt-cloud`. This section is required to have a `project-id` field with a valid project ID. 

Reviewer 2: Should they add this, or will it be there already?
```suggestion
4. In your `dbt_project.yml` file, add a section titled `dbt-cloud`. This section needs a `project-id` field with a valid project ID. 
```

```yaml
# dbt_project.yml
name:

version:
...

#IS IT CLOUD OR DBT-CLOUD PER ISSUE https://github.com/dbt-labs/docs.getdbt.com/issues/4014

> cloud: 

Reviewer: @greg-mckeon is it dbt-cloud or cloud? referencing issue https://github.com/dbt-labs/docs.getdbt.com/issues/4014

> - To find your project ID, go to **Develop** in the navigation menu. Select the dbt Cloud project URL, such as `https://cloud.getdbt.com/develop/26228/projects123456`, where the project ID is `123456`.

Reviewer 2: Suggest smoothing this flow a tiny bit. 
```suggestion
- To find your project ID, select **Develop** in the dbt Cloud navigation menu. You can use this URL to find the project ID. For example, in `https://cloud.getdbt.com/develop/26228/projects123456`, the project ID is `123456`.
```

## Use the dbt Cloud CLI

> The dbt Cloud CLI shares the same set of commands as dbt Core. When you invoke a dbt command, that command is sent to dbt Cloud for processing. 

Reviewer 2: Suggest making this sentence active.
```suggestion
The dbt Cloud CLI shares the same set of commands as dbt Core. dbt Cloud processes the dbt commands you invoke. 
```

The dbt Cloud CLI is a drop-in replacement for dbt Core. When you invoke a dbt command, that command is sent to dbt Cloud for processing. Since this is still invoking dbt under the hood, you'll still need to run `dbt deps`, followed by the same model build commands you typically run.

> 

Reviewer: @greg-mckeon are there any commands that can't work in the cloud cli? (e.g. dbt init?)

Author: Full list:

```sh
Available Commands:
  build         Run all seeds, models, snapshots, and tests in DAG order
  cancel        Cancel the most recent invocation
  clean         Delete all folders in the clean-targets list (usually the dbt_packages and target directories.)
  clone         Create clones of selected nodes based on their location in the manifest provided to --state.
  compile       Generates executable SQL from source, model, test and analysis files.
  deps          Pull the most recent version of the dependencies listed in packages.yml
  docs          Generate or serve the documentation website for your project
  help          Help about any command
  list          List the resources in your project
  parse         Parses the project and provides information on performance
  reattach      Reattach to the most recent invocation to retrieve logs and artifacts
  retry         Retry the nodes that failed in the previous run.
  run           Compile SQL and execute against the current target database.
  run-operation Run the named macro with any supplied arguments.
  seed          Load data from csv files into your data warehouse.
  show          Generates executable SQL for a named resource or inline query, runs that
SQL, and returns a preview of the results. Does not materialize anything to
the warehouse.
  snapshot      Execute snapshots defined in your project
  source        Manage your project's sources
  test          Runs tests on data in deployed models.
```

> The dbt Cloud CLI supports [project dependencies](/docs/collaborate/govern/project-dependencies), which is an exciting way to depend on another project using the metadata service in dbt Cloud. It instantly resolves references (or  `ref`) to public models defined in other projects. You don't need to execute or analyze these upstream models yourself. Instead, you treat them as an API that returns a dataset.

Reviewer 2: I tried to clarify "it" by changing it to dbt Cloud CLI, but should it be "project dependencies?" 
```suggestion
The dbt Cloud CLI supports [project dependencies](/docs/collaborate/govern/project-dependencies), which enables you to depend on another project using the metadata service in dbt Cloud. Project dependencies instantly resolves references (or  `ref`) to public models defined in other projects. You don't need to execute or analyze these upstream models yourself. Instead, you treat them as an API that returns a dataset.
```

Reviewer: ah yes it should be project dependencies. thanks for flagging this!

> Share feedback or request features you'd like to see on the [dbt community Slack](https://getdbt.slack.com/archives/C05M77P54FL).

Reviewer 2: ```suggestion
Share feedback or request features you'd like to see in the [dbt community Slack](https://getdbt.slack.com/archives/C05M77P54FL).
```

---
title: Configure dbt Cloud CLI
id: configure-cloud-cli
description: "Instructions on how to configure the dbt Cloud CLI"
sidebar_label: "Configure dbt Cloud CLI"
pagination_next: null
---

import CloudCLIFlag from '/snippets/_cloud-cli-flag.md';

<CloudCLIFlag/>

## Prerequisites

- You must set up a project in dbt Cloud.
- You must have your [personal development credentials](/docs/dbt-cloud-environments#set-developer-credentials) set for that project. The dbt Cloud CLI will use these credentials, stored securely in dbt Cloud, to communicate with your data platform.
- You must [enroll](/docs/dbt-versions/experimental-features) in the dbt Cloud beta features. 
	- To enroll, navigate to your **Profile Settings** and enable the **Beta** flag under **Experimental Features**.

## Configure the dbt Cloud CLI

Once you install the dbt Cloud CLI, you need to configure it to connect to a dbt Cloud project.

1. Ensure you meet the prerequisites above.

2. Download your credentials from dbt Cloud by clicking on the **Try the dbt Cloud CLI** banner on the dbt Cloud homepage.

> 3. Follow the banner instructions and download the config file to `~/.dbt/dbt_cloud.yml`. The config file looks like:

Author: This is correct for Mac and Linux, but should be: C:\Users\yourusername\\.dbt\dbt_cloud.yml on windows

---
title: Configure dbt Cloud CLI
id: configure-cloud-cli
description: "Instructions on how to configure the dbt Cloud CLI"
sidebar_label: "Configure dbt Cloud CLI"
pagination_next: null
---

import CloudCLIFlag from '/snippets/_cloud-cli-flag.md';

<CloudCLIFlag/>

## Prerequisites

- You must set up a project in dbt Cloud.
- You must have your [personal development credentials](/docs/dbt-cloud-environments#set-developer-credentials) set for that project. The dbt Cloud CLI will use these credentials, stored securely in dbt Cloud, to communicate with your data platform.
- You must [enroll](/docs/dbt-versions/experimental-features) in the dbt Cloud beta features. 
	- To enroll, navigate to your **Profile Settings** and enable the **Beta** flag under **Experimental Features**.

## Configure the dbt Cloud CLI

Once you install the dbt Cloud CLI, you need to configure it to connect to a dbt Cloud project.

1. Ensure you meet the prerequisites above.

> 2. Download your credentials from dbt Cloud by clicking on the **Try the dbt Cloud CLI** banner on the dbt Cloud homepage. Or you can download the credentials from the following links depending on your region:

Reviewer: ```suggestion
2. Download your credentials from dbt Cloud by clicking on the **Try the dbt Cloud CLI** banner on the dbt Cloud homepage. Alternatively, if you're in dbt Cloud, you can download the credentials from the links provided based on your region:
```
